src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/InboundEventsIntegrationTest.kt

```diff
@@ -89,6 +95,92 @@ class InboundEventsIntegrationTest : SqsIntegrationTestBase() {
+    )
+
+    val bookingId = webTestClient.createBooking(courtBookingRequest, COURT_USER)
+
+    videoBookingRepository.findById(bookingId).orElseThrow().statusCode isEqualTo StatusCode.ACTIVE
+
+    inboundEventsListener.onMessage(
+      raw(
+        PrisonerAppointmentsChangedEvent(
+          personReference = PersonReference(listOf(Identifier("NOMS", "123456"))),
+          additionalInformation = AppointmentsChangedInformation(
+            action = "YES",
+            prisonId = PENTONVILLE,
+            user = "SOME_USER",
+          ),
+        ),
+      ),
+    )
+
+    val persistedBooking = videoBookingRepository.findById(bookingId).orElseThrow().also { it.statusCode isEqualTo StatusCode.CANCELLED }
+
+    with(notificationRepository.findAll()) {
+      isPresent("b@b.com", ReleasedCourtBookingCourtEmail::class, persistedBooking)
+      isPresent("j@j.com", ReleasedCourtBookingCourtEmail::class, persistedBooking)
+      isPresent("g@g.com", ReleasedCourtBookingPrisonCourtEmail::class, persistedBooking)
+      isPresent("p@p.com", ReleasedCourtBookingPrisonCourtEmail::class, persistedBooking)
+    }
+  }
+
+  @Test
+  fun `should cancel a video booking and send transfer emails on receipt of a appointments changed event and last movement TRN`() {
+    videoBookingRepository.findAll() hasSize 0
+    prisonSearchApi().stubGetPrisoner("123456", prisonCode = PENTONVILLE, lastPrisonCode = PENTONVILLE, lastMovementTypeCode = "TRN")
+
+    val courtBookingRequest = courtBookingRequest(
+      courtCode = DERBY_JUSTICE_CENTRE,
+      prisonerNumber = "123456",
+      prisonCode = PENTONVILLE,
+      location = pentonvilleLocation,
+      date = tomorrow(),
+      startTime = LocalTime.of(9, 30),
+      endTime = LocalTime.of(20, 0),
+      comments = "integration test court booking comments",
+    )
+
+    val bookingId = webTestClient.createBooking(courtBookingRequest, COURT_USER)
+
+    videoBookingRepository.findById(bookingId).orElseThrow().statusCode isEqualTo StatusCode.ACTIVE
+
+    inboundEventsListener.onMessage(
+      raw(
+        PrisonerAppointmentsChangedEvent(
+          personReference = PersonReference(listOf(Identifier("NOMS", "123456"))),
+          additionalInformation = AppointmentsChangedInformation(
+            action = "YES",
+            prisonId = PENTONVILLE,
+            user = "SOME_USER",
+          ),
+        ),
+      ),
+    )
+
+    val persistedBooking = videoBookingRepository.findById(bookingId).orElseThrow().also { it.statusCode isEqualTo StatusCode.CANCELLED }
+
+    with(notificationRepository.findAll()) {
+      isPresent("b@b.com", TransferredCourtBookingCourtEmail::class, persistedBooking)
+      isPresent("j@j.com", TransferredCourtBookingCourtEmail::class, persistedBooking)
+      isPresent("g@g.com", TransferredCourtBookingPrisonCourtEmail::class, persistedBooking)
+      isPresent("p@p.com", TransferredCourtBookingPrisonCourtEmail::class, persistedBooking)
+    }
```

Comment:
Should there be a test for receiving the appointments-changed where the action is NO? i.e Don't cancel future bookings.